accessibility-checker-engine/src/v4/checker/Checker.ts

```diff
@@ -296,6 +301,39 @@ export class Checker implements IChecker {
         return retVal;
     }
 
+    private getReasonLevel(rsIds: string[], ruleId: string, reasonCode?: string) : eRulePolicy {
+        if (!rsIds) return eRulePolicy.INFORMATION; 
+        let rsInfo = this.ruleReasonLevels[ruleId];
+        let retVal = null;
+        if (rsIds) {
+            if (!(ruleId in this.ruleReasonLevels)) {
+                throw new Error("Rule triggered for which we have no rule level information "+ruleId);
+            }
+            for (const rsId of rsIds) {
+                if (rsId in rsInfo) {
+                    Object.keys(rsInfo[rsId]).forEach(code => { 
+                        let level = null;
+                        if (code === 'None')
+                            level = rsInfo[rsId]["None"];
+                        else if ((code.includes("--") && (code.includes(reasonCode+"--") || code.includes("--"+reasonCode))) || (!code.includes("--") && code.includes(reasonCode)))
```

Comment:
At the top of the function, define a regex: 
```const reCode = new RegExp(`(^|--)${reasonCode}($|--)`);```
Then, 
```suggestion
                        else if (reCode.test(code))
```

Reply:
neat change!